lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -24,6 +24,7 @@ def create
     result   = self.class.run(inst_cmd)
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:exitcode] == 0
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:errormessage].nil?
+    @resource.provider = self.class.instances.find{|prov| prov.name == @resource[:name]}
```

Comment:
So in order to do what I suggest, we would have to modify this method to accept a variable to inject into the erb: https://github.com/puppetlabs/puppetlabs-iis/blob/master/lib/puppet/provider/iis_powershell.rb#L149-L154. Is that possible?